presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -382,24 +397,34 @@ public static LongReadFunction timestampReadFunction()
      * {@link #timestampWriteFunction} instead.
      */
     @Deprecated
-    public static LongWriteFunction timestampWriteFunctionUsingSqlTimestamp()
+    public static LongWriteFunction timestampWriteFunctionUsingSqlTimestamp(TimestampType timestampType)
     {
-        return (statement, index, value) -> statement.setTimestamp(index, Timestamp.valueOf(fromPrestoTimestamp(value)));
+        checkArgument(timestampType.getPrecision() <= MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
+        return (statement, index, value) -> statement.setTimestamp(index, Timestamp.valueOf(fromPrestoTimestamp(timestampType, value)));
     }
 
-    public static LongWriteFunction timestampWriteFunction()
+    public static LongWriteFunction timestampWriteFunction(TimestampType timestampType)
     {
-        return (statement, index, value) -> statement.setObject(index, fromPrestoTimestamp(value));
+        checkArgument(timestampType.getPrecision() <= MAX_SHORT_PRECISION, "Precision is out of range: %s", timestampType.getPrecision());
+        return (statement, index, value) -> statement.setObject(index, fromPrestoTimestamp(timestampType, value));
     }
 
-    public static long toPrestoTimestamp(LocalDateTime localDateTime)
+    public static long toPrestoTimestamp(TimestampType timestampType, LocalDateTime localDateTime)
     {
-        return localDateTime.atZone(UTC).toInstant().toEpochMilli() * MICROSECONDS_PER_MILLISECOND;
+        long precision = timestampType.getPrecision();
+        checkArgument(precision <= MAX_SHORT_PRECISION, "Precision is out of range: %s", precision);
+        Instant instant = localDateTime.atZone(UTC).toInstant();
+        return instant.getEpochSecond() * MICROSECONDS_PER_SECOND + roundDiv(instant.getNano(), NANOSECONDS_PER_MICROSECOND);
```

Comment:
if timestampType.precision < 6, you need to round/truncate here

@martint what is the correct way to do this?
there are some utility methods in `io.prestosql.spi.type.Timestamps`, but I do not see something that's immediately applicable.